ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml

```diff
@@ -0,0 +1,699 @@
+        sshKey: |
+          ${SSH_PUB_KEY}
+        EOF
+        else
+            echo "Unsupported cluster type '${CLUSTER_TYPE}'"
+            exit 1
+        fi
+
+        TF_LOG=debug openshift-install --dir=/tmp/artifacts/installer create ignition-configs
+        mkdir -p /tmp/tf
+
+        cp /tmp/artifacts/installer/bootstrap.ign /srv
+
+        export BOOTSTRAP_URL="http://${JOB_NAME_SAFE}-bootstrap-exporter-${NAMESPACE}.svc.ci.openshift.org/bootstrap.ign"
+        export MASTER_IGN=$(cat /tmp/artifacts/installer/master.ign)
+        export WORKER_IGN=$(cat /tmp/artifacts/installer/worker.ign)
+
+        cat > /tmp/tf/terraform.tfvars <<-EOF
+          machine_cidr = "139.178.73.0/26"
+
+          vm_template = "rhcos-latest"
+
+          vsphere_cluster = "devel"
+
+          vsphere_datacenter = "dc1"
+
+          vsphere_datastore = "nvme-ds1"
+
+          vsphere_server = "vcsa-ci.vmware.devcluster.openshift.com"
+
+          ipam = "139.178.89.254"
+
+          cluster_id = "${CLUSTER_NAME}"
+
+          base_domain = "${BASE_DOMAIN}"
+
+          cluster_domain = "${CLUSTER_NAME}.${BASE_DOMAIN}"
+
+          bootstrap_ignition_url = "${BOOTSTRAP_URL}"
+
+          // Ignition config for the control plane machines. You should copy the contents of the master.ign generated by the installer.
+          control_plane_ignition = <<END_OF_MASTER_IGNITION
+          ${MASTER_IGN}
+          END_OF_MASTER_IGNITION
+
+          // Ignition config for the compute machines. You should copy the contents of the worker.ign generated by the installer.
+          compute_ignition = <<END_OF_WORKER_IGNITION
+          ${WORKER_IGN}
+          END_OF_WORKER_IGNITION
+        EOF
+
+        # Copy remaining tfvars
+        cp ${TFVARS_PATH} /tmp/tf
+
+        echo "Created ignition configs, waiting for infra to be created"
+        touch /tmp/ignition-configs
+
+        while true; do
+          if [[ -f /tmp/exit ]]; then
+            echo "Another process exited" 2>&1
+            exit 1
+          fi
+          if [[ ! -f /tmp/terraform-infra-success ]]; then
+            sleep 15 & wait
+            continue
+          fi
+          break
+        done
+
+        echo "Waiting for bootstrap to complete"
+        openshift-install --dir=/tmp/artifacts/installer wait-for bootstrap-complete
+        touch /tmp/bootstrap-complete
+        echo "Bootstrap complete, waiting for terraform step 3 to complete"
+
+        while true; do
+          if [[ -f /tmp/exit ]]; then
+            echo "Another process exited" 2>&1
+            exit 1
+          fi
+          if [[ ! -f /tmp/terraform-bootstrap-success ]]; then
+            sleep 15 & wait
+            continue
+          fi
+          break
+        done
+        echo "Completing UPI setup"
+        openshift-install --dir=/tmp/artifacts/installer wait-for finish &
+        wait "$!"
+
+    # Creates infra via terraform
+    - name: terraform
+      image: ${LOCAL_IMAGE_UPI_INSTALLER}
```

Comment:
Shouldn't this happen in the `create` container?  We'll have `openshift-install` inside `LOCAL_IMAGE_UPI_INSTALLER`, so I think one `setup` container using `LOCAL_IMAGE_UPI_INSTALLER` should be sufficient.

Reply:
Right, reworked this to use one container for installer and terraform